src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/DefaultComesLastCheck.java

```diff
@@ -77,11 +77,24 @@ public void visitToken(DetailAST ast) {
             final DetailAST lastGroupAST =
                 switchAST.getLastChild().getPreviousSibling();
 
-            if (defaultGroupAST.getLineNo() != lastGroupAST.getLineNo()
-                || defaultGroupAST.getColumnNo()
-                    != lastGroupAST.getColumnNo()) {
+            if (!containCaseBeforeOrAfterDefault(ast)
+                    && (defaultGroupAST.getLineNo() != lastGroupAST.getLineNo()
+                        || defaultGroupAST.getColumnNo()
+                            != lastGroupAST.getColumnNo())) {
                 log(ast, MSG_KEY);
             }
         }
     }
+
+    /**
+     * Examines ast to see if it is surrounded by a 'case' on either side.
+     *
+     * @param ast - root node.
+     * @return true if default is surrounded by a 'case' on either side or false.
+     */
+    private boolean containCaseBeforeOrAfterDefault(DetailAST ast) {
+        return ast.getPreviousSibling() != null
+                || ast.getNextSibling() != null
+                && ast.getNextSibling().getType() == TokenTypes.LITERAL_CASE;
```

Comment:
Thanks for handling this issue :smile_cat: This is just a simple question because I'm not that familiar with Checkstyle.

Is it possible that the previous sibling is a comment or anything else than a case?